libexec/tfenv-version-name

```diff
@@ -96,13 +96,23 @@ if [[ "${TFENV_VERSION}" =~ ^latest.*$ ]]; then
     | grep -e "${regex}" \
     | head -n 1)";
 
-  if [ -n "${version}" ]; then
-    log 'debug' "Version selected: ${version}";
-    TFENV_VERSION="${version}"
+  if [[ "${TFENV_AUTO_INSTALL:-true}" == "true" ]]; then
+    log 'debug' "Trying to find the remote version using the regex: ${regex}";
+    remote_version="$(tfenv-list-remote | grep -e "${regex}" | head -n 1)"
+    if [[ -n "${remote_version}" ]]; then
+        if [[ "${version}" != "${remote_version}" ]]; then
+          log 'debug' "The installed version '${version}' does not much the remote version '${remote_version}'";
+          TFENV_VERSION="${remote_version}"
+        else
+          TFENV_VERSION="${version}"
+        fi
+    else
+      log 'error' "No versions matching '${requested}' found in remote";
+    fi
   else
-    log 'error' "No installed versions of terraform matched '${TFENV_VERSION}'";
-  fi;
-
+    [[ -n "${version}" ]] && TFENV_VERSION="${version}" \
```

Comment:
this used to be a cleaner if/else and i think this codebase prefers explicit if/else for actions but allows `condition && log 'info' 'ok' || log 'error' 'not ok' pattern.